src/meta-srv/src/selector/load_based.rs

```diff
@@ -43,22 +47,44 @@ impl Selector for LoadBasedSelector {
         let stat_keys: Vec<StatKey> = lease_kvs.keys().map(|k| k.into()).collect();
         let stat_kvs = ctx.meta_peer_client.get_dn_stat_kvs(stat_keys).await?;
 
+        let leader_peer_ids = if let (Some(catalog), Some(schema), Some(table)) =
+            (&ctx.catalog, &ctx.schema, &ctx.table)
+        {
+            let table_metadata_manager =
+                TableMetadataManager::new(KvBackendAdapter::wrap(ctx.kv_store.clone()));
+
+            let table_name = table_metadata_manager
+                .table_name_manager()
+                .get(TableNameKey::new(catalog, schema, table))
+                .await
+                .context(error::TableMetadataManagerSnafu)?;
+
+            if let Some(table_name) = table_name {
+                let route = table_metadata_manager
+                    .table_route_manager()
+                    .get(table_name.table_id())
+                    .await
+                    .context(error::TableMetadataManagerSnafu)?
+                    .with_context(|| error::TableRouteNotFoundSnafu {
```

Comment:
Is it possible that table route value is legitimate to be not found here? For example, the table is dropped while executing this selector.

Reply:
No, the Drop table will remove all metadata in a single txn.

Reply:
But it's possible the table route value is not there when you retrieve it here. After all, you are not fetching the table name key and table route key at the same time.